framework/base/Model.php

```diff
@@ -239,9 +239,15 @@ public function scenarios()
      */
     public function formName()
     {
-        $reflector = new ReflectionClass($this);
+        static $formName = null;
```

Comment:
@laszlovl So wouldn't an alternative be a static class method?

```
protected static $_var;

public function getFoo()
{
    if (static::$_var === null) {
        static::$_var = ...
    }
    return static::$_var
}
```

By using `static::$_var` instead of `self::$_var` it would even allow child classes to keep their own values (at least that's what I think it does). Maybe also a feasible approach?